hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/TriggerAction.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.api.model.trigger;
+
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+/**
+ * Link an ActionDefinition with a Trigger.
+ *
+ * It can add optional constraints that determine when an action will be executed:
+ *
+ * - A set of Alert.Status (represented by its string value).
+ *   The action will be executed if the Alert which is linked is on one of the states defined.
+ *   This is not applicable if the action is linked with an Event.
```

Comment:
Perhaps replace "This is not applicable if the action is linked with an Event" with:

"Unlike Alerts, Events don't have lifecycle, TriggerActions on Events are all executed at Event creation time."